src/nvim/eval.c

```diff
@@ -5360,7 +5360,7 @@ bool garbage_collect(bool testing)
 
   // history items (ShaDa additional elements)
   if (p_hi) {
-    for (uint8_t i = 0; i < HIST_COUNT; i++) {
+    for (int i = 0; i < HIST_COUNT; i++) {
```

Comment:
A cast to `uint8_t` may also be needed below.
```suggestion
    for (HistoryType i = 0; i < HIST_COUNT; i++) {
```